### PR TITLE
Reorder rubygems includes to ensure its early upgrade.

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

**Comment by ksubrama** [comment]
Needs https://github.com/chef/omnibus-software/pull/483 to go in.  Then remove the branch specified in the Gemfile before merging.

File: config/software/chef-provisioning-fog.rb [hunk 1]
```diff
@@ -24,11 +24,13 @@
...
 else
   dependency "ruby"
-  dependency "rubygems"
 end
 
-dependency "bundler"
+dependency "rubygems"
 dependency "nokogiri"
+dependency "bundler"
+dependency "chef"
```
Comment (jkeiser) [inline config/software/chef-provisioning-fog.rb]: Why the extra dependency?

File: Gemfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
 gem 'omnibus', github: 'chef/omnibus'
-gem 'omnibus-software', github: 'chef/omnibus-software'
+gem 'omnibus-software', github: 'chef/omnibus-software', branch: 'ksubrama/rubygems'
```
Comment (schisamo) [inline Gemfile]: I assume we'll move back to master before merging?

File: config/software/chef-provisioning-fog.rb (hunk 1 above)
Comment (ksubrama) [inline config/software/chef-provisioning-fog.rb]: I made every plugin depend on chef to ensure that we never install a plugin or its dependencies before we install all of chef.

File: Gemfile (hunk 2 above)
Comment (ksubrama) [inline Gemfile]: Yep - already removed.

**Pull request merged**